tests/unit/customizations/autoprompt/test_history.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import mock
+import os
+import shutil
+import tempfile
+
+from prompt_toolkit.buffer import Buffer
+from prompt_toolkit.completion import Completion
+from prompt_toolkit.document import Document
+from prompt_toolkit.history import History
+from prompt_toolkit.formatted_text import FormattedText
+
+from awscli.customizations.autoprompt.history import (
+    HistoryCompleter, HistoryDriver)
+from awscli.testutils import unittest, temporary_file
+
+
+class TestHistoryCompleter(unittest.TestCase):
+    def setUp(self):
+        self.strings = [
+            'chime associate-phone-number-with-user',
+            'opsworks-cm associate-node',
+            'deploy add-tags-to-on-premises-instances',
+            's3 ls',
+        ]
+        self.buffer = mock.Mock(spec=Buffer)
+        history = mock.Mock(spec=History)
+        history.get_strings = mock.Mock()
```

Comment:
Do we need to assign `mock.Mock()` here? The method should already be mocked out and can just immediately assign a `return_value` to it.